### PR TITLE
设计模式：模块编辑

### PR DESCRIPTION
**Pull request by nikogu**
First of all, thanks for your contribution! :-)

Please makes sure these boxes are checked before submitting your PR, thank you!
- [x] Make sure you follow antd's [code convention](https://github.com/ant-design/ant-design/wiki/Code-convention-for-antd).
- [x] Run `npm run lint` and fix those errors before submitting in order to keep consistent code style.
- [x] Rebase before creating a PR to keep commit history clear.
- [x] Add some descriptions and refer relative issues for you PR.

### CODE REVIEW

File: docs/pattern/table/demo/edit-modular.md [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+      this.editCache[index] = { ...this.state.dataSource[index] };
+    }
+
+    this.editCache[index][key] = value;
+  }
+
+  render() {
+    const props = this.props;
+
+    // 和历史动态修改的所以需要放到render中
```
Comment (nikogu) [inline docs/pattern/table/demo/edit-modular.md]: 这个地方写错别字了

File: docs/pattern/table/demo/edit-modular.md [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+    }
+
+    if (nextProps.customerList !== this.props.customerList) {
+      this.setState({
+        customerList: nextProps.customerList
+      });
+    }
+  }
+
+  changeEditState(type, index) {
```
Comment (saiyagg) [inline docs/pattern/table/demo/edit-modular.md]: https://github.com/ant-design/ant-design/pull/1258

Comment (nikogu) [inline docs/pattern/table/demo/edit-modular.md]: @saiyagg 没bind

File: docs/pattern/table/demo/edit-modular.md [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+];
+
+const mockCustomerList = ['竹卢', '乐仪', '水鱼', '徐干', '左宜'];
+
+ReactDOM.render(
+  <EditableTable
+    dataSource={ mockData }
+    customerList={ mockCustomerList }
+    onChange={ (oldData, newData) => console.log(oldData, newData) }
+  />);
```
Comment (saiyagg) [inline docs/pattern/table/demo/edit-modular.md]: render 函数没 mountNode 参赛

**Comment by afc163** [comment]
最好起一个有意义的分支名。

File: docs/pattern/table/demo/edit-modular.md (hunk 2 above)
Comment (benjycui) [inline docs/pattern/table/demo/edit-modular.md]: @afc163 这个应该不建议用户用 ES7?

File: docs/pattern/table/demo/edit-modular.md [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+    }
+
+    this.editCache[index][key] = value;
+  }
+
+  render() {
+    const props = this.props;
+
+    // 和历史动态修改的所以需要放到render中
+    let columns = [
```
Comment (benjycui) [inline docs/pattern/table/demo/edit-modular.md]: `columns` 放 constructor？然后应该用 `const` 吧。。

File: docs/pattern/table/demo/edit-modular.md (hunk 2 above)
Comment (nikogu) [inline docs/pattern/table/demo/edit-modular.md]: @saiyagg 据说，这样写jslint通不过

**Pull request merged**